SIMPLE/src/edu/manipal/icas/simple/impl/views/ProfileCreationViewImpl.java

```diff
@@ -0,0 +1,438 @@
+		verifyAndContinueButton = new JButton("Verify email and continue");
+		confirmPasswordPasswordField = new JPasswordField();
+		continueButton = new JButton("Continue");
+		nameTextField = new JTextField();
+		genderTextField = new JTextField();
+		contactNumberTextField = new JTextField();
+		dateSettings = new DatePickerSettings();
+		dateSettings.setFirstDayOfWeek(DayOfWeek.SUNDAY);
+		datePicker = new DatePicker(dateSettings);
+		nextButton = new JButton("Next");
+		passportOfficeComboBox = new JComboBox();
+		finishButton = new JButton("Create citizen profile");
+		initializeUi();
+	}
+
+	/**
+	 * Initializes the profile creation UI and populates it with the create profile
+	 * icon and the product name label.
+	 */
+	private void initializeUi() {
+		JPanel container = new JPanel();
+		container.setLayout(new GridBagLayout());
+		GridBagConstraints c = new GridBagConstraints();
+		c.fill = GridBagConstraints.HORIZONTAL;
+		c.gridheight = 1;
+		c.gridwidth = 1;
+		c.gridx = 0;
+		c.gridy = 0;
+		container.add(new JLabel(), c);
+
+		c.gridheight = 1;
+		c.gridwidth = 1;
+		c.gridx = 1;
+		c.gridy = 0;
+		c.ipady = 15;
+		container.add(new JLabel(ResourceConstants.IMAGE_CREATE_CITIZEN_PROFILE_ICON), c);
+
+		c.gridheight = 1;
+		c.gridwidth = 1;
+		c.ipady = 0;
+		c.gridx = 2;
+		c.gridy = 0;
+		container.add(new JLabel(), c);
+
+		JLabel productNameLabel = new JLabel(StringConstants.PRODUCT_NAME, JLabel.CENTER);
+		productNameLabel.setFont(ResourceConstants.FONT_HEADING_BOLD);
+		c.gridx = 1;
+		c.gridy = 1;
+		c.gridheight = 1;
+		c.gridwidth = 1;
+		container.add(productNameLabel, c);
+
+		JLabel newProfileLabel = new JLabel("Create a new profile", JLabel.CENTER);
+		newProfileLabel.setFont(ResourceConstants.FONT_SUBHEADING_PLAIN);
+		c.gridx = 1;
+		c.gridy = 2;
+		c.gridheight = 1;
+		c.gridwidth = 1;
+		container.add(newProfileLabel, c);
+
+		c.gridy = 3;
+		c.ipady = 15;
+		container.add(new JLabel(), c);
+
+		c.gridheight = 120;
+		c.weighty = 1;
+		c.gridy = 4;
+		c.anchor = GridBagConstraints.PAGE_START;
+
+		container.add(initialiseStepsContainerPanel(), c);
+
+		setIconImage(ResourceConstants.IMAGE_CREATE_CITIZEN_PROFILE_ICON.getImage());
+		container.setBackground(new Color(255, 255, 255));
+		add(container);
+		setSize(new Dimension(300, 400));
+		setLocationRelativeTo(null);
+		setResizable(false);
+
+		addWindowListener(new WindowAdapter() {
+			@Override
+			public void windowClosing(WindowEvent e) {
+				// TODO Auto-generated method stub
+				super.windowClosing(e);
+				System.exit(0);
+			}
+		});
+	}
+
+	/**
+	 * Initialises the steps container panel which contains only one panel at a
+	 * time.
+	 * 
+	 * @return the steps conatiner panel
+	 */
+	private JPanel initialiseStepsContainerPanel() {
+		JPanel stepsContainerPanel = new JPanel();
+		stepsContainerPanel.setLayout(new BoxLayout(stepsContainerPanel, BoxLayout.LINE_AXIS));
+		stepsContainerPanel.add(initialiseStep1Panel());
```

Comment:
The panel that's generated here is not the same one that's returned in L#349. Perhaps you could create a class member for these and return it in the getter instead.

Same with all other step panels and the step container.

Reply:
👍Ok